packages/react-components/lib/snapshot-control-button-group-reducer.ts

```diff
@@ -0,0 +1,57 @@
+export type ActionFormat<T, K = undefined> = {
+  type: T;
+  payload: K;
+};
+
+export enum SnapshotActionType {
+  ADD_CONTENT = 'addContent',
+  CLEAR = 'clear',
+  RESTORE = 'restore',
+}
+
+export enum SnapshotStateType {
+  CONTENT = 'content',
+}
+
+export type SnapshotAction =
+  | ActionFormat<'addContent', any>
+  | ActionFormat<'clear', any>
+  | ActionFormat<'restore', any>;
+
+export type SnapshotState = {
+  [SnapshotStateType.CONTENT]: any;
+};
+
+function mergeContent(
+  currentContent: any,
+  storedContent: any,
+  replaceCurrentContent: boolean = true,
+) {
+  const newContents = Object.assign({}, currentContent);
+  Object.keys(storedContent).forEach((element) => {
+    if (replaceCurrentContent) newContents[element] = storedContent[element];
+    else {
+      // If the element is already on the currentContent do not replace it
+      if (!(element in currentContent)) newContents[element] = storedContent[element];
+    }
+  });
+  return newContents;
+}
```

Comment:
Since this is a general state or reducer, should we keep it with `any`? or we should add the possible types with `|`?